cmake/flash/CMakeLists.txt

```diff
@@ -130,7 +130,7 @@ endif()
 
 zephyr_get(WEST_DIR)
 if(WEST_DIR)
-  set(WEST "PYTHONPATH=${WEST_DIR}/src" "${PYTHON_EXECUTABLE};${WEST_DIR}/src/west/app/main.py;--zephyr-base=${ZEPHYR_BASE} ")
+  set(WEST "PYTHONPATH=${WEST_DIR}/src${SEP}$ENV{PYTHONPATH}" "${PYTHON_EXECUTABLE};${WEST_DIR}/src/west/app/main.py;--zephyr-base=${ZEPHYR_BASE} ")
```

Comment:
Note, `SEP` is undefined here.

We need to ensure that it's correctly defined on all platforms.
I propose creating a zephyr extension function for this, so that it can be fetched when needed and in the correct form (genex / configure time) usage.

For example
```
zephyr_host(SEPARATOR <out-var> [GENEX])
```
so that if `GENEX` is provided, then `$<SEMICOLON>` is returned on windows and `:` on other OSs.
if `GENEX` is not provided, then `\\;` is returned on windows and `:` on other OSs.

Currently in this PR, `SEP` is undefined, hence code will not work as expected.
CMake trace output:
```
52529 .../zephyr/cmake/flash/CMakeLists.txt(133):  set(WEST PYTHONPATH=/projects/github/west//src/projects/github/ncs/zephyr/scripts /usr/bin/python3.8;/projects/github/west//src/west/app/main.py;--zephyr-base=/projects/github/ncs/zephyr  )
```
notice how there is no separator or space between the `/west//src` and following `/projects/...` paths.

Reasons for using a function and not CMake variables is:
- avoid collision on `SEP` variable name as it is quite generic.
- Clear distinction when separator is needed for configure time vs generator time.
- Easy replacement of function with `cmake -E env --modify <NAME>:path_list_append=<VALUE>` when CMake >=3.25 is required.